sycl/test-e2e/syclcompat/util/util_find_first_set.cpp

```diff
@@ -104,16 +104,16 @@ void test_find_first_set() {
   sycl::queue q_ct1 = *dev_ct1.default_queue();
   int *test_result, host_test_result = 0;
 
-  test_result = sycl::malloc_shared<int>(sizeof(int), q_ct1);
-  *test_result = 0;
+  test_result = sycl::malloc_device<int>(1, q_ct1);
+  q_ct1.memcpy(test_result, &host_test_result, sizeof(int)).wait();
 
   q_ct1.parallel_for(
       sycl::nd_range<3>(sycl::range<3>(1, 1, 1), sycl::range<3>(1, 1, 1)),
       [=](sycl::nd_item<3> item_ct1) { find_first_set_test(test_result); });
 
   dev_ct1.queues_wait_and_throw();
+  q_ct1.memcpy(&host_test_result, test_result, sizeof(int)).wait();
   find_first_set_test(&host_test_result);
-  assert(*test_result == 0);
```

Comment:
Changes for syclcompat look great thanks. 
The only missing thing is this assert that checks the output of the device code, which was accidentally removed. 
See that the result is copied to `host_test_result` and that is then overwritten before the assert. Those are two different results that need to be checked.

Reply:
Ah yes, thanks for catching that!

Reply:
@victor-eds Take a look when you have some time please, Basically I've addedd another assert that checks whether the host value is zero and then  to check that the device value is zero memcpy transfers the device value to host and checks that is zero since we cannot directly access device ptr from the host.

Reply:
That also works. Thanks for addressing it